src/eduid_userdb/proofing/proofing_element.py

```diff
@@ -0,0 +1,171 @@
+#        with the distribution.
+#     3. Neither the name of the NORDUnet nor the names of its
+#        contributors may be used to endorse or promote products derived
+#        from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+
+import copy
+
+from eduid_userdb.element import Element, VerifiedElement, _update_something_by, _update_something_ts
+from eduid_userdb.exceptions import UserDBValueError
+
+
+__author__ = 'lundberg'
+
+
+class NinProofingElement(VerifiedElement):
+    """
+    Elements that can be verified or not.
+
+    Properties of NinProofingElement:
+
+        number
+        created_by
+        created_ts
+        is_verified
+        verified_by
+        verified_ts
+        verification_code
+
+    :param data: element parameters from database
+
+    :type data: dict
+    """
+    def __init__(self, number=None, application=None, created_ts=None, verified=False, verification_code=None,
+                 data=None):
+
+        data_in = data
+        data = copy.copy(data_in)  # to not modify callers data
+
+        if data is None:
+            if created_ts is None:
+                created_ts = True
+            data = dict(number=number,
+                        created_by=application,
+                        created_ts=created_ts,
+                        verified=verified,
+                        verification_code=verification_code,
+                        )
+
+        VerifiedElement.__init__(self, data)
+        self.number = data.pop('number')
+
+    @property
+    def number(self):
+        """
+        This is the nin number.
+
+        :return: nin number.
+        :rtype: str | unicode
+        """
+        return self._data['number']
+
+    @number.setter
+    def number(self, value):
+        """
+        :param value: nin number.
+        :type value: str | unicode
+        """
+        if not isinstance(value, basestring):
+            raise UserDBValueError("Invalid 'number': {!r}".format(value))
+        self._data['number'] = str(value.lower())
+
+    def to_dict(self, old_userdb_format=False):
+        res = super(NinProofingElement, self).to_dict(old_userdb_format)
```

Comment:
Maybe we shouldn't have old_userdb_format for new things that never existed pre-userdb?
In this particular case, this parameter is a no-op once it reaches Element.to_dict().